File: test/py/case36_stateless_value.py
# 値型最適化候補のテスト（インスタンス状態なし）。


class Tag:
    def id(self) -> int:
        return 7


def run_case36_stateless_value() -> None:
    t = Tag()
    u = t
    print(t.id())
    print(u.id())


if __name__ == "__main__":
    run_case36_stateless_value()

File: test/cpp/case36_stateless_value.cpp
#include "cpp_module/py_runtime.h"

// 値型最適化候補のテスト（インスタンス状態なし）。


struct Tag {
    int64 id() {
        return 7;
    }
};

void run_case36_stateless_value() {
    Tag t = Tag();
    Tag u = t;
    py_print(t.id());
    py_print(u.id());
}

int main(int argc, char** argv) {
    pytra_configure_from_argv(argc, argv);
    run_case36_stateless_value();
    return 0;
}
